src/metabase/events/view_log.clj

```diff
@@ -7,21 +7,51 @@
    [metabase.models.query.permissions :as query-perms]
    [metabase.public-settings.premium-features :as premium-features]
    [metabase.util :as u]
+   [metabase.util.grouper :as grouper]
    [metabase.util.log :as log]
    [methodical.core :as m]
    [steffan-westcott.clj-otel.api.trace.span :as span]
    [toucan2.core :as t2]))
 
-(defn increment-view-counts!
-  "Increments the view_count column for a model given a list of ids.
-   Assumes the model has one primary key `id`, and the column for the view count is named `view_count`"
-  [_model & _ids]
-  ;; Disable view_count updates to handle perf issues  (for now) (#44359)
-  #_
-  (when (seq ids)
-    (t2/query {:update (t2/table-name model)
-               :set    {:view_count [:+ :view_count [:inline 1]]}
-               :where  [:in :id ids]})))
+(defn- group-by-frequency
+  "Given a list of items, returns a map of frequencies to items.
+    (group-by-frequency [:a :a :b :b :c :c :c])
+    ;; => {2 [:a :b] 3 [:c]}"
```

Comment:
nit: this code actually builds lists of the items now, and they are in the opposite order. you could use `(fnil conj [])` if you want this behavior still.

Reply:
I see that we've used this `fnil` in `metabot` as well. I've often made a util for this called `conjv` because I think it's pretty useful.

Reply:
That's a very nice helper. I took the liberty to add this to `metabase.util`. The credit will always belong to you :p